### DISCUSSION TITLE
The command will prompt "Unknown Language 'xml' for xxxxx, supported Languages are java"

### DISCUSSION THREAD

**Discussion by kakakpy** [pmd/pmd#4258] [Q&A]
Based on the detection rule of an xml file created by AbstractXmlRule, the unit test is ok, and the maven dependency is added: pmd-xml. But when using the pmd command line
The command will prompt "Unknown Language 'xml' for xxxxx, supported Languages are java"

**Comment by adangel** [comment]
Can you be a bit more specific in what you actually did?
Judging from the error, you are missing the dependency to "pmd-xml" somewhere.
Did you use https://github.com/pmd/pmd-examples/tree/java as a starting point? Then make sure, you have added the dependency to "pmd-xml" in your xxx-dist-module (e.g. in the sample, it is here: https://github.com/pmd/pmd-examples/blob/java/pmd-java-dist/pom.xml). This xxx-dist-module is creating the distribution which contains the command line tool.

**Comment by kakakpy** [comment]
What you provide is executable. My packaging is jar-with-dependencies. I customize mainclass and can only do java checks. Why not xml? How do I pack it

**Reply by kakakpy** [reply]
@adangel Looking forward to your answer

**Reply by adangel** [reply]
Please provide your pom.xml file you use to create the jar-with-dependencies jar file.
Otherwise I can just repeat "you are missing the dependency to "pmd-xml" somewhere.".
